src/Tools/Program.cs

```diff
@@ -0,0 +1,337 @@
+                {
+                    foreach (var projectOrSolution in projectsOrSolutions)
+                    {
+                        if (projectOrSolution.EndsWith(".sln", StringComparison.OrdinalIgnoreCase))
+                        {
+                            await computeSolutionMetricDataAsync(workspace, projectOrSolution).ConfigureAwait(false);
+                        }
+                        else
+                        {
+                            Debug.Assert(projectOrSolution.EndsWith(".csproj", StringComparison.OrdinalIgnoreCase) ||
+                                projectOrSolution.EndsWith(".vbproj", StringComparison.OrdinalIgnoreCase));
+                            await computeProjectMetricDataAsync(workspace, projectOrSolution).ConfigureAwait(false);
+                        }
+                    }
+                }
+
+                return (builder.ToImmutable(), ErrorCode.None);
+            }
+            catch (Exception ex)
+            {
+                Console.Write(ex.Message);
+                return (ImmutableArray<(string, CodeAnalysisMetricData)>.Empty, ErrorCode.ComputeException);
+            }
+
+            async Task computeProjectMetricDataAsync(MSBuildWorkspace workspace, string projectFile)
+            {
+                if (!quiet)
+                {
+                    Console.WriteLine($"Loading {Path.GetFileName(projectFile)}...");
+                }
+
+                var project = await workspace.OpenProjectAsync(projectFile, CancellationToken.None).ConfigureAwait(false);
+
+                if (!quiet)
+                {
+                    Console.WriteLine($"Computing code metrics for {Path.GetFileName(projectFile)}...");
+                }
+
+                var metricData = await CodeAnalysisMetricData.ComputeAsync(project, CancellationToken.None).ConfigureAwait(false);
+                builder.Add((projectFile, metricData));
+            }
+
+            async Task computeSolutionMetricDataAsync(MSBuildWorkspace workspace, string solutionFile)
+            {
+                if (!quiet)
+                {
+                    Console.WriteLine($"Loading {Path.GetFileName(solutionFile)}...");
+                }
+
+                var solution = await workspace.OpenSolutionAsync(solutionFile, CancellationToken.None).ConfigureAwait(false);
+
+                if (!quiet)
+                {
+                    Console.WriteLine($"Computing code metrics for {Path.GetFileName(solutionFile)}...");
+                }
+
+                foreach (var project in solution.Projects)
+                {
+                    if (!quiet)
+                    {
+                        Console.WriteLine($"    Computing code metrics for {Path.GetFileName(project.FilePath)}...");
+                    }
+
+                    var metricData = await CodeAnalysisMetricData.ComputeAsync(project, CancellationToken.None).ConfigureAwait(false);
```

Comment:
where CodeAnalysisMetricData type came from?

Reply:
This is existing type we use for Code metrics computation for our ported Code metrics analyzer.

Reply:
so it is in feature package? is feature package has IVT to this tool?

Reply:
No it is checked into the Analyzer.Utilities project in this repo: https://github.com/dotnet/roslyn-analyzers/blob/master/src/Utilities/CodeMetrics/CodeAnalysisMetricData.cs

Reply:
I added reference to Features package as `MSBuildWorkspace.Create` was attempting to load C# and VB and Core Features assemblies.